oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/MapAggregator.java

```diff
@@ -220,17 +220,15 @@ private MapAggregator(MapAggregator<U, ?> obj, MapReducer<IndexValuePair<U, X>>
       );
     } else {
       MapAggregator<OSHDBCombinedIndex<U, V>, ? extends OSHDBMapReducible> ret;
-      if (this.mapReducer.forClass.equals(OSMContribution.class)) {
-        ret = this.flatMap(x -> gs.splitOSMContribution((OSMContribution) x).entrySet())
+      if (mapReducer.isContributionViewQuery()) {
+        ret = this.flatMap(x -> gs.splitOSMContribution((OSMContributionImpl) x).entrySet())
             .aggregateBy(Entry::getKey, geometries.keySet()).map(Entry::getValue);
-      } else if (this.mapReducer.forClass.equals(OSMEntitySnapshot.class)) {
-        ret = this.flatMap(x -> gs.splitOSMEntitySnapshot((OSMEntitySnapshot) x).entrySet())
+      } else if (mapReducer.isOSMEntitySnapshotQuery()) {
+        ret = this.flatMap(x -> gs.splitOSMEntitySnapshot((OSMEntitySnapshotImpl) x).entrySet())
```

Comment:
Use the interface instead of the implementing class for the cast.

```suggestion
        ret = this.flatMap(x -> gs.splitOSMEntitySnapshot((OSMEntitySnapshot) x).entrySet())
```